1-react/store/slomux.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export const createStore = (reducer, initialState) => {
+  let currentState = initialState
+  const listeners = []
+
+  const getState = () => currentState
+  const dispatch = action => {
+    currentState = reducer(currentState, action)
+    listeners.forEach(listener => listener())
+  }
+
+  const subscribe = listener => listeners.push(listener)
+
+  return { getState, dispatch, subscribe }
+}
+
+export const connect = (mapStateToProps, mapDispatchToProps) =>
+  Component => {
+    class WrappedComponent extends React.Component {
+      render() {
+        return (
+          <Component
+            {...this.props}
+            {...mapStateToProps(this.context.store.getState(), this.props)}
+            {...mapDispatchToProps(this.context.store.dispatch, this.props)}
+          />
+        )
+      }
+
+      componentDidUpdate() {
```

Comment:
Из документации React:

> componentDidUpdate не вызывается при первом рендере.

Т.е. компонент не сможет получать обновления стора до тех пор, пока в нём не произойдёт повторный рендер.

Хуже того, `componentDidUpdate` будет вызываться после _каждого_ обновления, и я не думаю, что офромлять подписку нужно более одного раза.

Давай заменим это на [componentDidMount](https://ru.reactjs.org/docs/react-component.html#componentdidmount) =)

```suggestion
      componentDidMount() {
```